src/contentmanager.cpp

```diff
@@ -943,12 +945,26 @@ bool ContentManager::handleZimFileInMonitoredDirLogged(QString dir, QString file
     return status == MonitoredZimFileInfo::ADDED_TO_THE_LIBRARY;
 }
 
+bool ContentManager::MonitoredZimFileInfo::fileKeepsBeingModified() const
+{
+    // A file is considered stable if it has stayed unchanged for at least
+    // this long.
+    const qint64 FILE_STABILITY_DURATION_MS = 1000;
+
+    const QDateTime now = QDateTime::currentDateTime();
+    return this->lastModified > now.addMSecs(-FILE_STABILITY_DURATION_MS);
+}
+
 void ContentManager::MonitoredZimFileInfo::updateStatus(const MonitoredZimFileInfo& prevInfo)
 {
     Q_ASSERT(prevInfo.status != ADDED_TO_THE_LIBRARY);
 
     if ( this->lastModified == prevInfo.lastModified ) {
         this->status = UNCHANGED_KNOWN_BAD_ZIM_FILE;
+    } else if ( prevInfo.status == PROCESS_LATER ) {
+        this->status = DEFERRED_PROCESSING_ALREADY_PENDING;
```

Comment:
I'm not sure to properly understand the status evolution of a zim being downloaded (behind our back)

`DEFERRED_PROCESSING_ALREADY_PENDING` is set only here and never read. So I suppose it is just a status to not set something else (which may trigger different action).

But when we go out of this status ? Is it handleZimFileInMonitoredDirDeferred which change it ?
But it changes it to PROCESS_NOW and so we add to the library "right now". Should we check again that file has been modified for a second ?

Reply:
I believe `deferHandlingOfZimFileInMonitoredDir` is called first which sets the timer of 1 second, then `handleZimFileInMonitoredDirDeferred` is called to set it to PROCESS_NOW.

Reply:
I believe `deferHandlingOfZimFileInMonitoredDir` is called first which sets the timer of 1 second, then `handleZimFileInMonitoredDirDeferred` is called to set it to PROCESS_NOW.

Reply:
> `DEFERRED_PROCESSING_ALREADY_PENDING` is set only here and never read. So I suppose it is just a status to not set something else (which may trigger different action).

Mostly yes (though `DEFERRED_PROCESSING_ALREADY_PENDING` is read implicitly by the logging code, but that is only for debugging).

> But when we go out of this status ? Is it handleZimFileInMonitoredDirDeferred which change it ? But it changes it to PROCESS_NOW and so we add to the library "right now". Should we check again that file has been modified for a second ?

This status is set only on a temporary variable and is never recorded in directory monitoring's state (`m_knownZimsInDir`). The same is true for the `PROCESS_NOW`, `BEING_DOWNLOADED_BY_US`, and `UNCHANGED_KNOWN_BAD_ZIM_FILE` statuses, however, I guess with them the situation is less confusing since they do not imply any delayed action - a decision how to proceed is made immediately (in `ContentManager::handleZimFileInMonitoredDir()`). In a sense, that is also the case with `DEFERRED_PROCESSING_ALREADY_PENDING` - as soon as it is discovered that a deferred action is already registered, we simply stop processing the current event. From that point of view `BEING_DOWNLOADED_BY_US`, `UNCHANGED_KNOWN_BAD_ZIM_FILE` and `DEFERRED_PROCESSING_ALREADY_PENDING` could be reduced to a single value `NOTHING_TO_BE_DONE` and exist merely for logging/debugging purposes.